### PR TITLE
Expose interface and enum in selectableOption for combobox

### LINKED ISSUE #2091: Combobox: should expose API used by Combobox options
### Bug Report

- __Package version(s)__: latest master

#### Priorities and help requested (not applicable if asking question):

Requested priority: (High) 

#### Describe the issue:

The ComboBox requires options whose type is **ISelectableOption** which is not exposed in typescript.

In sp-client, I try to import them directly
```
import {
  ISelectableOption,
  SelectableOptionMenuItemType
} from 'office-ui-fabric-react/lib/utilities/selectableOption/SelectableOption.Props';
```
But it then throws warnings like this:
Warning - [webpack] 'dist':
C:/Users/liaye/GitRepos/sp-client-backup/common/temp/~/office-ui-fabric-react/lib/utilities/selectableOption/SelectableOption.Props.js
Fabric path "/office-ui-fabric-react/lib/utilities/selectableOption/SelectableOption.Props.js" does not correspond to a well-defined export. Large portions of office-ui-fabric-react may be unintentionally bundled. This request originated in "C:\Users\liaye\GitRepos\sp-client-backup\apps\sp-pages\lib\core\pageLayouts\events\CategoryCombobox.js".

### PR DESCRIPTION
**Pull request by ipip2005**
#### Pull request checklist

- [ #2091 ] Addresses an existing issue: Fixes #2091 
- [ done ] Include a change request file using `$ npm run change`

#### Description of changes

The API used by ComboBox props "options" is `ISelectableOption` which is not exposed yet from fabric-react. Directly import it from lib will cause warnings blocking a merge in sp-client.

### CODE REVIEW

**Comment by ipip2005** [comment]
@jspurlin  @dzearing  Could you help review the change? It just exposes the APIs for ComboBox consumers.

File: packages/office-ui-fabric-react/src/Utilities.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export * from '@uifabric/utilities/lib/index';
+export * from './utilities/selectableOption';
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/Utilities.ts]: Hmm. Shouldn't this just be exposed in ComboBox?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/Utilities.ts]: Unless we move the private utilities in oufr to the actual utilities library, which may not be unreasonable, I would suggest we just export with the top level imports that need it.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/Utilities.ts]: E.g. in components/DetailsList/index.ts, you can see we have a number of exports:

```ts
export * from '../../utilities/selection/index';
export * from '../GroupedList/GroupedList.Props';
export * from './DetailsList';
export * from './DetailsList.Props';
export * from './DetailsRow';
export * from './DetailsRowCheck';
```

**Comment by ipip2005** [comment]
Addressed @dzearing

**Comment by ipip2005** [comment]
@dzearing do you have any other concerns after my commit? We'd like to push this fix soon so we might be able to consume it in next fabric-react version bumping up in sp-client

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts [hunk 2]
```diff
@@ -2,11 +2,13 @@ import { IIconProps } from '../../Icon';
 import { ISelectableOption } from '../../utilities/selectableOption/SelectableOption.Props';
 import { ISelectableDroppableTextProps } from '../../utilities/selectableOption/SelectableDroppableText.Props';
 
+export * from '../../utilities/selectableOption/SelectableOption.Props';
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: why not follow what `Dropdown` does and have something like 
```
export interface IComboBoxOption extends ISelectableOption { }
```

and then `options?: IComboBoxOption`?

Comment (ipip2005) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: Thanks, looks better for extensibility in future

File: packages/office-ui-fabric-react/src/Utilities.ts (hunk 1 above)
Comment (ipip2005) [inline packages/office-ui-fabric-react/src/Utilities.ts]: Thanks for the suggestion, fixed

**Pull request merged**